deps/verifier/src/az_snp_vtpm/mod.rs

```diff
@@ -188,6 +206,57 @@ fn verify_snp_report(
     Ok(())
 }
 
+/// Verifies the signature of the attestation report using the provided certificate chain and vendor certificates.
+fn verify_report_signature(report: &AttestationReport, vcek: &Vcek) -> Result<()> {
+    // OpenSSL bindings do not expose custom extensions
+    // Parse the key using x509_parser
+
+    let endorsement_key_der = &vcek.0.to_der()?;
+    let parsed_endorsement_key = X509Certificate::from_der(endorsement_key_der)?
+        .1
+        .tbs_certificate;
+
+    let common_name = get_common_name(&vcek.0).context("No common name found in certificate")?;
+
+    // if the common name is "VCEK", then the key is a VCEK
+    // so lets check the chip id
+    if common_name == "VCEK"
+        && get_oid_octets::<64>(&parsed_endorsement_key, HW_ID_OID)? != report.chip_id
+    {
+        bail!("Chip ID mismatch");
+    }
+
+    // tcb version
+    // these integer extensions are 3 bytes with the last byte as the data
+    if get_oid_int(&parsed_endorsement_key, UCODE_SPL_OID)? != report.reported_tcb.microcode {
+        return Err(anyhow!("Microcode version mismatch"));
+    }
+
+    if get_oid_int(&parsed_endorsement_key, SNP_SPL_OID)? != report.reported_tcb.snp {
+        return Err(anyhow!("SNP version mismatch"));
+    }
+
+    if get_oid_int(&parsed_endorsement_key, TEE_SPL_OID)? != report.reported_tcb.tee {
+        return Err(anyhow!("TEE version mismatch"));
+    }
+
+    if get_oid_int(&parsed_endorsement_key, LOADER_SPL_OID)? != report.reported_tcb.bootloader {
+        return Err(anyhow!("Boot loader version mismatch"));
+    }
+
+    // verify report signature
+    let sig = ecdsa::EcdsaSig::try_from(&report.signature)?;
+    let data = &bincode::serialize(&report)?[..=0x29f];
```

Comment:
Here we have a hardcoded `0x29f`, probably it is good to add a note in code it means the [AttestationReport](https://docs.rs/sev/6.1.0/sev/firmware/guest/struct.AttestationReport.html) struct except signature field, or use [memoffset](https://crates.io/crates/memoffset) crate to generate the value.